cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/image/AzureImageFilter.java

```diff
@@ -72,4 +82,19 @@ private Image removeNonMarketplaceRegions(Image image) {
 
         return new Image(image, filteredMap);
     }
+
+    private Image removeMarketplaceRegions(Image image) {
+        Map<String, Map<String, String>> filteredMap = image.getImageSetsByProvider().entrySet()
+                .stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, computeRegionMapWithoutMarketplace()));
+
+        return new Image(image, filteredMap);
+    }
+
+    @NotNull
+    private static Function<Map.Entry<String, Map<String, String>>, Map<String, String>> computeRegionMapWithoutMarketplace() {
```

Comment:
- is static really required?
- I think it would help a lot with readability if it would be a simple method instead of a `Function`